// @passageidentity/passage-node/lib/cjs/generated/models/AuthMethods.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * Passage Management API
 * Passage\'s management API to manage your Passage apps and users.
 *
 * The version of the OpenAPI document: 1
 * Contact: support@passage.id
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.AuthMethodsToJSON = exports.AuthMethodsFromJSONTyped = exports.AuthMethodsFromJSON = exports.instanceOfAuthMethods = void 0;
const MagicLinkAuthMethod_1 = require("./MagicLinkAuthMethod");
const OtpAuthMethod_1 = require("./OtpAuthMethod");
const PasskeysAuthMethod_1 = require("./PasskeysAuthMethod");
/**
 * Check if a given object implements the AuthMethods interface.
 */
function instanceOfAuthMethods(value) {
    let isInstance = true;
    isInstance = isInstance && "passkeys" in value;
    isInstance = isInstance && "otp" in value;
    isInstance = isInstance && "magic_link" in value;
    return isInstance;
}
exports.instanceOfAuthMethods = instanceOfAuthMethods;
function AuthMethodsFromJSON(json) {
    return AuthMethodsFromJSONTyped(json, false);
}
exports.AuthMethodsFromJSON = AuthMethodsFromJSON;
function AuthMethodsFromJSONTyped(json, ignoreDiscriminator) {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        'passkeys': (0, PasskeysAuthMethod_1.PasskeysAuthMethodFromJSON)(json['passkeys']),
        'otp': (0, OtpAuthMethod_1.OtpAuthMethodFromJSON)(json['otp']),
        'magic_link': (0, MagicLinkAuthMethod_1.MagicLinkAuthMethodFromJSON)(json['magic_link']),
    };
}
exports.AuthMethodsFromJSONTyped = AuthMethodsFromJSONTyped;
function AuthMethodsToJSON(value) {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        'passkeys': (0, PasskeysAuthMethod_1.PasskeysAuthMethodToJSON)(value.passkeys),
        'otp': (0, OtpAuthMethod_1.OtpAuthMethodToJSON)(value.otp),
        'magic_link': (0, MagicLinkAuthMethod_1.MagicLinkAuthMethodToJSON)(value.magic_link),
    };
}
exports.AuthMethodsToJSON = AuthMethodsToJSON;
