// @passageidentity/passage-node/lib/cjs/generated/models/MagicLinkAuthMethod.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * Passage Management API
 * Passage\'s management API to manage your Passage apps and users.
 *
 * The version of the OpenAPI document: 1
 * Contact: support@passage.id
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.MagicLinkAuthMethodToJSON = exports.MagicLinkAuthMethodFromJSONTyped = exports.MagicLinkAuthMethodFromJSON = exports.instanceOfMagicLinkAuthMethod = void 0;
const TtlDisplayUnit_1 = require("./TtlDisplayUnit");
/**
 * Check if a given object implements the MagicLinkAuthMethod interface.
 */
function instanceOfMagicLinkAuthMethod(value) {
    let isInstance = true;
    isInstance = isInstance && "enabled" in value;
    isInstance = isInstance && "ttl" in value;
    isInstance = isInstance && "ttl_display_unit" in value;
    return isInstance;
}
exports.instanceOfMagicLinkAuthMethod = instanceOfMagicLinkAuthMethod;
function MagicLinkAuthMethodFromJSON(json) {
    return MagicLinkAuthMethodFromJSONTyped(json, false);
}
exports.MagicLinkAuthMethodFromJSON = MagicLinkAuthMethodFromJSON;
function MagicLinkAuthMethodFromJSONTyped(json, ignoreDiscriminator) {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        'enabled': json['enabled'],
        'ttl': json['ttl'],
        'ttl_display_unit': (0, TtlDisplayUnit_1.TtlDisplayUnitFromJSON)(json['ttl_display_unit']),
    };
}
exports.MagicLinkAuthMethodFromJSONTyped = MagicLinkAuthMethodFromJSONTyped;
function MagicLinkAuthMethodToJSON(value) {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        'enabled': value.enabled,
        'ttl': value.ttl,
        'ttl_display_unit': (0, TtlDisplayUnit_1.TtlDisplayUnitToJSON)(value.ttl_display_unit),
    };
}
exports.MagicLinkAuthMethodToJSON = MagicLinkAuthMethodToJSON;
