// @passageidentity/passage-node/lib/cjs/generated/models/MagicLinkChannel.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * Passage Management API
 * Passage\'s management API to manage your Passage apps and users.
 *
 * The version of the OpenAPI document: 1
 * Contact: support@passage.id
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.MagicLinkChannelToJSON = exports.MagicLinkChannelFromJSONTyped = exports.MagicLinkChannelFromJSON = exports.MagicLinkChannel = void 0;
/**
 *
 * @export
 */
exports.MagicLinkChannel = {
    Email: 'email',
    Phone: 'phone'
};
function MagicLinkChannelFromJSON(json) {
    return MagicLinkChannelFromJSONTyped(json, false);
}
exports.MagicLinkChannelFromJSON = MagicLinkChannelFromJSON;
function MagicLinkChannelFromJSONTyped(json, ignoreDiscriminator) {
    return json;
}
exports.MagicLinkChannelFromJSONTyped = MagicLinkChannelFromJSONTyped;
function MagicLinkChannelToJSON(value) {
    return value;
}
exports.MagicLinkChannelToJSON = MagicLinkChannelToJSON;
