// @passageidentity/passage-node/lib/cjs/generated/models/OtpAuthMethod.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * Passage Management API
 * Passage\'s management API to manage your Passage apps and users.
 *
 * The version of the OpenAPI document: 1
 * Contact: support@passage.id
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.OtpAuthMethodToJSON = exports.OtpAuthMethodFromJSONTyped = exports.OtpAuthMethodFromJSON = exports.instanceOfOtpAuthMethod = void 0;
const TtlDisplayUnit_1 = require("./TtlDisplayUnit");
/**
 * Check if a given object implements the OtpAuthMethod interface.
 */
function instanceOfOtpAuthMethod(value) {
    let isInstance = true;
    isInstance = isInstance && "enabled" in value;
    isInstance = isInstance && "ttl" in value;
    isInstance = isInstance && "ttl_display_unit" in value;
    return isInstance;
}
exports.instanceOfOtpAuthMethod = instanceOfOtpAuthMethod;
function OtpAuthMethodFromJSON(json) {
    return OtpAuthMethodFromJSONTyped(json, false);
}
exports.OtpAuthMethodFromJSON = OtpAuthMethodFromJSON;
function OtpAuthMethodFromJSONTyped(json, ignoreDiscriminator) {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        'enabled': json['enabled'],
        'ttl': json['ttl'],
        'ttl_display_unit': (0, TtlDisplayUnit_1.TtlDisplayUnitFromJSON)(json['ttl_display_unit']),
    };
}
exports.OtpAuthMethodFromJSONTyped = OtpAuthMethodFromJSONTyped;
function OtpAuthMethodToJSON(value) {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        'enabled': value.enabled,
        'ttl': value.ttl,
        'ttl_display_unit': (0, TtlDisplayUnit_1.TtlDisplayUnitToJSON)(value.ttl_display_unit),
    };
}
exports.OtpAuthMethodToJSON = OtpAuthMethodToJSON;
