// @passageidentity/passage-node/lib/cjs/generated/models/PasskeysAuthMethod.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * Passage Management API
 * Passage\'s management API to manage your Passage apps and users.
 *
 * The version of the OpenAPI document: 1
 * Contact: support@passage.id
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.PasskeysAuthMethodToJSON = exports.PasskeysAuthMethodFromJSONTyped = exports.PasskeysAuthMethodFromJSON = exports.instanceOfPasskeysAuthMethod = void 0;
/**
 * Check if a given object implements the PasskeysAuthMethod interface.
 */
function instanceOfPasskeysAuthMethod(value) {
    let isInstance = true;
    isInstance = isInstance && "enabled" in value;
    return isInstance;
}
exports.instanceOfPasskeysAuthMethod = instanceOfPasskeysAuthMethod;
function PasskeysAuthMethodFromJSON(json) {
    return PasskeysAuthMethodFromJSONTyped(json, false);
}
exports.PasskeysAuthMethodFromJSON = PasskeysAuthMethodFromJSON;
function PasskeysAuthMethodFromJSONTyped(json, ignoreDiscriminator) {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        'enabled': json['enabled'],
    };
}
exports.PasskeysAuthMethodFromJSONTyped = PasskeysAuthMethodFromJSONTyped;
function PasskeysAuthMethodToJSON(value) {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        'enabled': value.enabled,
    };
}
exports.PasskeysAuthMethodToJSON = PasskeysAuthMethodToJSON;
