// @passageidentity/passage-node/lib/cjs/generated/models/TtlDisplayUnit.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * Passage Management API
 * Passage\'s management API to manage your Passage apps and users.
 *
 * The version of the OpenAPI document: 1
 * Contact: support@passage.id
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.TtlDisplayUnitToJSON = exports.TtlDisplayUnitFromJSONTyped = exports.TtlDisplayUnitFromJSON = exports.TtlDisplayUnit = void 0;
/**
 * Deprecated Property. The preferred unit for displaying the TTL. This value is for display only.
 * * `s` - seconds
 * * `m` - minutes
 * * `h` - hours
 * * `d` - days
 * @export
 */
exports.TtlDisplayUnit = {
    S: 's',
    M: 'm',
    H: 'h',
    D: 'd'
};
function TtlDisplayUnitFromJSON(json) {
    return TtlDisplayUnitFromJSONTyped(json, false);
}
exports.TtlDisplayUnitFromJSON = TtlDisplayUnitFromJSON;
function TtlDisplayUnitFromJSONTyped(json, ignoreDiscriminator) {
    return json;
}
exports.TtlDisplayUnitFromJSONTyped = TtlDisplayUnitFromJSONTyped;
function TtlDisplayUnitToJSON(value) {
    return value;
}
exports.TtlDisplayUnitToJSON = TtlDisplayUnitToJSON;
