// @passageidentity/passage-node/lib/cjs/generated/models/UpdateOtpAuthMethod.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * Passage Management API
 * Passage\'s management API to manage your Passage apps and users.
 *
 * The version of the OpenAPI document: 1
 * Contact: support@passage.id
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.UpdateOtpAuthMethodToJSON = exports.UpdateOtpAuthMethodFromJSONTyped = exports.UpdateOtpAuthMethodFromJSON = exports.instanceOfUpdateOtpAuthMethod = void 0;
const runtime_1 = require("../runtime");
const TtlDisplayUnit_1 = require("./TtlDisplayUnit");
/**
 * Check if a given object implements the UpdateOtpAuthMethod interface.
 */
function instanceOfUpdateOtpAuthMethod(value) {
    let isInstance = true;
    return isInstance;
}
exports.instanceOfUpdateOtpAuthMethod = instanceOfUpdateOtpAuthMethod;
function UpdateOtpAuthMethodFromJSON(json) {
    return UpdateOtpAuthMethodFromJSONTyped(json, false);
}
exports.UpdateOtpAuthMethodFromJSON = UpdateOtpAuthMethodFromJSON;
function UpdateOtpAuthMethodFromJSONTyped(json, ignoreDiscriminator) {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        'enabled': !(0, runtime_1.exists)(json, 'enabled') ? undefined : json['enabled'],
        'ttl': !(0, runtime_1.exists)(json, 'ttl') ? undefined : json['ttl'],
        'ttl_display_unit': !(0, runtime_1.exists)(json, 'ttl_display_unit') ? undefined : (0, TtlDisplayUnit_1.TtlDisplayUnitFromJSON)(json['ttl_display_unit']),
    };
}
exports.UpdateOtpAuthMethodFromJSONTyped = UpdateOtpAuthMethodFromJSONTyped;
function UpdateOtpAuthMethodToJSON(value) {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        'enabled': value.enabled,
        'ttl': value.ttl,
        'ttl_display_unit': (0, TtlDisplayUnit_1.TtlDisplayUnitToJSON)(value.ttl_display_unit),
    };
}
exports.UpdateOtpAuthMethodToJSON = UpdateOtpAuthMethodToJSON;
