// @passageidentity/passage-node/lib/cjs/generated/models/UpdatePasskeysAuthMethod.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * Passage Management API
 * Passage\'s management API to manage your Passage apps and users.
 *
 * The version of the OpenAPI document: 1
 * Contact: support@passage.id
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.UpdatePasskeysAuthMethodToJSON = exports.UpdatePasskeysAuthMethodFromJSONTyped = exports.UpdatePasskeysAuthMethodFromJSON = exports.instanceOfUpdatePasskeysAuthMethod = void 0;
const runtime_1 = require("../runtime");
/**
 * Check if a given object implements the UpdatePasskeysAuthMethod interface.
 */
function instanceOfUpdatePasskeysAuthMethod(value) {
    let isInstance = true;
    return isInstance;
}
exports.instanceOfUpdatePasskeysAuthMethod = instanceOfUpdatePasskeysAuthMethod;
function UpdatePasskeysAuthMethodFromJSON(json) {
    return UpdatePasskeysAuthMethodFromJSONTyped(json, false);
}
exports.UpdatePasskeysAuthMethodFromJSON = UpdatePasskeysAuthMethodFromJSON;
function UpdatePasskeysAuthMethodFromJSONTyped(json, ignoreDiscriminator) {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        'enabled': !(0, runtime_1.exists)(json, 'enabled') ? undefined : json['enabled'],
    };
}
exports.UpdatePasskeysAuthMethodFromJSONTyped = UpdatePasskeysAuthMethodFromJSONTyped;
function UpdatePasskeysAuthMethodToJSON(value) {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        'enabled': value.enabled,
    };
}
exports.UpdatePasskeysAuthMethodToJSON = UpdatePasskeysAuthMethodToJSON;
