// @passageidentity/passage-node/lib/cjs/generated/models/UserSocialConnections.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * Passage Management API
 * Passage\'s management API to manage your Passage apps and users.
 *
 * The version of the OpenAPI document: 1
 * Contact: support@passage.id
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.UserSocialConnectionsToJSON = exports.UserSocialConnectionsFromJSONTyped = exports.UserSocialConnectionsFromJSON = exports.instanceOfUserSocialConnections = void 0;
const runtime_1 = require("../runtime");
const GithubSocialConnection_1 = require("./GithubSocialConnection");
const GoogleSocialConnection_1 = require("./GoogleSocialConnection");
/**
 * Check if a given object implements the UserSocialConnections interface.
 */
function instanceOfUserSocialConnections(value) {
    let isInstance = true;
    return isInstance;
}
exports.instanceOfUserSocialConnections = instanceOfUserSocialConnections;
function UserSocialConnectionsFromJSON(json) {
    return UserSocialConnectionsFromJSONTyped(json, false);
}
exports.UserSocialConnectionsFromJSON = UserSocialConnectionsFromJSON;
function UserSocialConnectionsFromJSONTyped(json, ignoreDiscriminator) {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        'google': !(0, runtime_1.exists)(json, 'google') ? undefined : (0, GoogleSocialConnection_1.GoogleSocialConnectionFromJSON)(json['google']),
        'github': !(0, runtime_1.exists)(json, 'github') ? undefined : (0, GithubSocialConnection_1.GithubSocialConnectionFromJSON)(json['github']),
    };
}
exports.UserSocialConnectionsFromJSONTyped = UserSocialConnectionsFromJSONTyped;
function UserSocialConnectionsToJSON(value) {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        'google': (0, GoogleSocialConnection_1.GoogleSocialConnectionToJSON)(value.google),
        'github': (0, GithubSocialConnection_1.GithubSocialConnectionToJSON)(value.github),
    };
}
exports.UserSocialConnectionsToJSON = UserSocialConnectionsToJSON;
