// @passageidentity/passage-node/lib/cjs/generated/models/UserStatus.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * Passage Management API
 * Passage\'s management API to manage your Passage apps and users.
 *
 * The version of the OpenAPI document: 1
 * Contact: support@passage.id
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.UserStatusToJSON = exports.UserStatusFromJSONTyped = exports.UserStatusFromJSON = exports.UserStatus = void 0;
/**
 *
 * @export
 */
exports.UserStatus = {
    Active: 'active',
    Inactive: 'inactive',
    Pending: 'pending'
};
function UserStatusFromJSON(json) {
    return UserStatusFromJSONTyped(json, false);
}
exports.UserStatusFromJSON = UserStatusFromJSON;
function UserStatusFromJSONTyped(json, ignoreDiscriminator) {
    return json;
}
exports.UserStatusFromJSONTyped = UserStatusFromJSONTyped;
function UserStatusToJSON(value) {
    return value;
}
exports.UserStatusToJSON = UserStatusToJSON;
