// @passageidentity/passage-node/lib/esm/generated/models/AuthMethods.js
/* tslint:disable */
/* eslint-disable */
/**
 * Passage Management API
 * Passage\'s management API to manage your Passage apps and users.
 *
 * The version of the OpenAPI document: 1
 * Contact: support@passage.id
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
import { MagicLinkAuthMethodFromJSON, MagicLinkAuthMethodToJSON, } from './MagicLinkAuthMethod';
import { OtpAuthMethodFromJSON, OtpAuthMethodToJSON, } from './OtpAuthMethod';
import { PasskeysAuthMethodFromJSON, PasskeysAuthMethodToJSON, } from './PasskeysAuthMethod';
/**
 * Check if a given object implements the AuthMethods interface.
 */
export function instanceOfAuthMethods(value) {
    let isInstance = true;
    isInstance = isInstance && "passkeys" in value;
    isInstance = isInstance && "otp" in value;
    isInstance = isInstance && "magic_link" in value;
    return isInstance;
}
export function AuthMethodsFromJSON(json) {
    return AuthMethodsFromJSONTyped(json, false);
}
export function AuthMethodsFromJSONTyped(json, ignoreDiscriminator) {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        'passkeys': PasskeysAuthMethodFromJSON(json['passkeys']),
        'otp': OtpAuthMethodFromJSON(json['otp']),
        'magic_link': MagicLinkAuthMethodFromJSON(json['magic_link']),
    };
}
export function AuthMethodsToJSON(value) {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        'passkeys': PasskeysAuthMethodToJSON(value.passkeys),
        'otp': OtpAuthMethodToJSON(value.otp),
        'magic_link': MagicLinkAuthMethodToJSON(value.magic_link),
    };
}
