// @passageidentity/passage-node/lib/esm/generated/models/GithubSocialConnection.js
/* tslint:disable */
/* eslint-disable */
/**
 * Passage Management API
 * Passage\'s management API to manage your Passage apps and users.
 *
 * The version of the OpenAPI document: 1
 * Contact: support@passage.id
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
/**
 * Check if a given object implements the GithubSocialConnection interface.
 */
export function instanceOfGithubSocialConnection(value) {
    let isInstance = true;
    isInstance = isInstance && "provider_id" in value;
    isInstance = isInstance && "created_at" in value;
    isInstance = isInstance && "last_login_at" in value;
    isInstance = isInstance && "provider_identifier" in value;
    return isInstance;
}
export function GithubSocialConnectionFromJSON(json) {
    return GithubSocialConnectionFromJSONTyped(json, false);
}
export function GithubSocialConnectionFromJSONTyped(json, ignoreDiscriminator) {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        'provider_id': json['provider_id'],
        'created_at': (new Date(json['created_at'])),
        'last_login_at': (new Date(json['last_login_at'])),
        'provider_identifier': json['provider_identifier'],
    };
}
export function GithubSocialConnectionToJSON(value) {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        'provider_id': value.provider_id,
        'created_at': (value.created_at.toISOString()),
        'last_login_at': (value.last_login_at.toISOString()),
        'provider_identifier': value.provider_identifier,
    };
}
