// @passageidentity/passage-node/lib/esm/generated/models/OtpAuthMethod.js
/* tslint:disable */
/* eslint-disable */
/**
 * Passage Management API
 * Passage\'s management API to manage your Passage apps and users.
 *
 * The version of the OpenAPI document: 1
 * Contact: support@passage.id
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
import { TtlDisplayUnitFromJSON, TtlDisplayUnitToJSON, } from './TtlDisplayUnit';
/**
 * Check if a given object implements the OtpAuthMethod interface.
 */
export function instanceOfOtpAuthMethod(value) {
    let isInstance = true;
    isInstance = isInstance && "enabled" in value;
    isInstance = isInstance && "ttl" in value;
    isInstance = isInstance && "ttl_display_unit" in value;
    return isInstance;
}
export function OtpAuthMethodFromJSON(json) {
    return OtpAuthMethodFromJSONTyped(json, false);
}
export function OtpAuthMethodFromJSONTyped(json, ignoreDiscriminator) {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        'enabled': json['enabled'],
        'ttl': json['ttl'],
        'ttl_display_unit': TtlDisplayUnitFromJSON(json['ttl_display_unit']),
    };
}
export function OtpAuthMethodToJSON(value) {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        'enabled': value.enabled,
        'ttl': value.ttl,
        'ttl_display_unit': TtlDisplayUnitToJSON(value.ttl_display_unit),
    };
}
