// @passageidentity/passage-node/lib/esm/generated/models/PasskeysAuthMethod.js
/* tslint:disable */
/* eslint-disable */
/**
 * Passage Management API
 * Passage\'s management API to manage your Passage apps and users.
 *
 * The version of the OpenAPI document: 1
 * Contact: support@passage.id
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
/**
 * Check if a given object implements the PasskeysAuthMethod interface.
 */
export function instanceOfPasskeysAuthMethod(value) {
    let isInstance = true;
    isInstance = isInstance && "enabled" in value;
    return isInstance;
}
export function PasskeysAuthMethodFromJSON(json) {
    return PasskeysAuthMethodFromJSONTyped(json, false);
}
export function PasskeysAuthMethodFromJSONTyped(json, ignoreDiscriminator) {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        'enabled': json['enabled'],
    };
}
export function PasskeysAuthMethodToJSON(value) {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        'enabled': value.enabled,
    };
}
