// @passageidentity/passage-node/lib/esm/generated/models/UpdateOtpAuthMethod.js
/* tslint:disable */
/* eslint-disable */
/**
 * Passage Management API
 * Passage\'s management API to manage your Passage apps and users.
 *
 * The version of the OpenAPI document: 1
 * Contact: support@passage.id
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
import { exists } from '../runtime';
import { TtlDisplayUnitFromJSON, TtlDisplayUnitToJSON, } from './TtlDisplayUnit';
/**
 * Check if a given object implements the UpdateOtpAuthMethod interface.
 */
export function instanceOfUpdateOtpAuthMethod(value) {
    let isInstance = true;
    return isInstance;
}
export function UpdateOtpAuthMethodFromJSON(json) {
    return UpdateOtpAuthMethodFromJSONTyped(json, false);
}
export function UpdateOtpAuthMethodFromJSONTyped(json, ignoreDiscriminator) {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        'enabled': !exists(json, 'enabled') ? undefined : json['enabled'],
        'ttl': !exists(json, 'ttl') ? undefined : json['ttl'],
        'ttl_display_unit': !exists(json, 'ttl_display_unit') ? undefined : TtlDisplayUnitFromJSON(json['ttl_display_unit']),
    };
}
export function UpdateOtpAuthMethodToJSON(value) {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        'enabled': value.enabled,
        'ttl': value.ttl,
        'ttl_display_unit': TtlDisplayUnitToJSON(value.ttl_display_unit),
    };
}
