// @passageidentity/passage-node/lib/esm/generated/models/UpdatePasskeysAuthMethod.js
/* tslint:disable */
/* eslint-disable */
/**
 * Passage Management API
 * Passage\'s management API to manage your Passage apps and users.
 *
 * The version of the OpenAPI document: 1
 * Contact: support@passage.id
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
import { exists } from '../runtime';
/**
 * Check if a given object implements the UpdatePasskeysAuthMethod interface.
 */
export function instanceOfUpdatePasskeysAuthMethod(value) {
    let isInstance = true;
    return isInstance;
}
export function UpdatePasskeysAuthMethodFromJSON(json) {
    return UpdatePasskeysAuthMethodFromJSONTyped(json, false);
}
export function UpdatePasskeysAuthMethodFromJSONTyped(json, ignoreDiscriminator) {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        'enabled': !exists(json, 'enabled') ? undefined : json['enabled'],
    };
}
export function UpdatePasskeysAuthMethodToJSON(value) {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        'enabled': value.enabled,
    };
}
