// @passageidentity/passage-node/lib/esm/generated/models/UserSocialConnections.js
/* tslint:disable */
/* eslint-disable */
/**
 * Passage Management API
 * Passage\'s management API to manage your Passage apps and users.
 *
 * The version of the OpenAPI document: 1
 * Contact: support@passage.id
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
import { exists } from '../runtime';
import { GithubSocialConnectionFromJSON, GithubSocialConnectionToJSON, } from './GithubSocialConnection';
import { GoogleSocialConnectionFromJSON, GoogleSocialConnectionToJSON, } from './GoogleSocialConnection';
/**
 * Check if a given object implements the UserSocialConnections interface.
 */
export function instanceOfUserSocialConnections(value) {
    let isInstance = true;
    return isInstance;
}
export function UserSocialConnectionsFromJSON(json) {
    return UserSocialConnectionsFromJSONTyped(json, false);
}
export function UserSocialConnectionsFromJSONTyped(json, ignoreDiscriminator) {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        'google': !exists(json, 'google') ? undefined : GoogleSocialConnectionFromJSON(json['google']),
        'github': !exists(json, 'github') ? undefined : GithubSocialConnectionFromJSON(json['github']),
    };
}
export function UserSocialConnectionsToJSON(value) {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        'google': GoogleSocialConnectionToJSON(value.google),
        'github': GithubSocialConnectionToJSON(value.github),
    };
}
